.github/workflows/linkcheck.yml

```diff
@@ -0,0 +1,44 @@
+name: LinkCheck - All files 🔗
+
+on:
+  schedule:
+    - cron: "0 5 * * 0" # Runs at 05:00 on Sunday.
+  workflow_dispatch:
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
```

Comment:
Would it be too time-consuming to run this CI for every PullReqest as said in https://github.com/pyvista/pyvista/issues/1574#issuecomment-1206871455, if we run it for every PullReqest?
It would be helpful to trigger it with a PullRequest since it would be a review of an invalid link.
The document builds running in parallel take about the same amount of time, so it may not affect the overall CI completion time.